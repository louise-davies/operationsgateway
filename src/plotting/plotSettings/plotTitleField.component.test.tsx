import React from 'react';
import PlotTitleField from './plotTitleField.component';
import { render, screen } from '@testing-library/react';
import type { RenderResult } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Plot Title Field', () => {
  let user;
  const changePlotTitle = jest.fn();

  const createView = (): RenderResult => {
    return render(<PlotTitleField changePlotTitle={changePlotTitle} />);
  };

  beforeEach(() => {
    user = userEvent.setup({ delay: null });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { asFragment } = createView();

    expect(asFragment()).toMatchSnapshot();
  });

  it('lets user change the plot title and calls changePlotTitle', async () => {
    createView();

    const titleInput = screen.getByRole('textbox', { name: 'Title' });

    await user.type(titleInput, 'Test title');

    expect(titleInput).toHaveValue('Test title');

    expect(changePlotTitle).toHaveBeenCalledWith('Test title');
  });
});
