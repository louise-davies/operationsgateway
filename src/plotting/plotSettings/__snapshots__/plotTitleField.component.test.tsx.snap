// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plot Title Field renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      for=":r0:"
      id=":r0:-label"
      style="font-size: 12px;"
    >
      Title
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      style="font-size: 12px;"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r0:"
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-1ftyaf0"
        >
          <span>
            Title
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</DocumentFragment>
`;
